### PR TITLE
fix(l1): solve block-building-related bugs in order to pass hive engine tests

### PR DESCRIPTION
**Pull request by fmoletta**
**Motivation**
Implements necessary fixes across the codebase and adds the now-passing hive tests to the CI. As most of this fixes are small, I grouped them in the same PR.

**Description**
Fixes:
* Add field `shouldOverrideBuilder` to `engine_getPayload` rpc response
* Fix blob_gas_used computation for rpc receipts
* Fix blob_gas_price computation for rpc receipts
* Set a max size for limbdbx DB (1TB) to avoid MapFull errors when storing multiple blobs
* Return `InvalidPayloadAttributes` instead of `BadParams` when receiving an invalid `PayloadAttributes` param in `ForkChoiceUpdated`
* Return `UnsupportedFork` error for pre-Cancun payload attributes (forkChoiceUpdated)
* Fix error returned when parsing NewPayloadV3Request (BadParams -> WrongParam)

Tests added to CI:
* `Blob Transactions On Block 1, Cancun Genesis`
* `Blob Transactions On Block 1, Shanghai Genesis`
* `Blob Transaction Ordering, Single Account, Single Blob`
* `Blob Transaction Ordering, Single Account, Dual Blob`
* `Parallel Blob Transactions`
* `ForkchoiceUpdatedV3 Modifies Payload ID on Different Beacon Root`
* `NewPayloadV3 After Cancun` (3 tests)
* `NewPayloadV3 Versioned Hashes` (18 tests)
* `cancun/Unique Payload ID` (6 tests)
* `cancun/ParentHash equals BlockHash on NewPayload` (2 tests)
* `cancun/RPC` (4 tests)
* `ForkchoiceUpdated Version on Payload Request`
*  `cancun/Re-Execute Payload`
* `cancun/Payload Build after New Invalid Payload` (2 tests)
* `cancun/Build Payload with Invalid ChainID` (3 tests)

Tests that may need some looking into out of the scope of this PR:
* Tests failing due to triggering re-orgs (we should check if this is the expected test behaviour):
  * `Incorrect BlobGasUsed` (2 tests) 
  * `Inconsistent Head in ForkchoiceState`
  * `Inconsistent Safe in ForkchoiceState`
  * `Inconsistent Finalized in ForkchoiceState`
* Tests failing due to triggering a sync (we should check if this is the expected test behaviour):
  * `In-Order Consecutive Payload Execution`
* Uncategorized:
  * `NewPayload with Missing ForkchoiceUpdated`
  * `Invalid PayloadAttributes`  (multiple) mostly adding checks
  * `InvalidNewPayload` (multiple)

Tests failing due to known issues:
* ` Replace Blob Transactions` should be fixed by #681 
* `Request Blob Pooled Transactions` (2 tests)  requires #384 
* `Bad Hash on NewPayload` (8 tests): failing due to sync not being implemented (sync milestone)
* `Unknown HeadBlockHash`: requires sync
* `Multiple New Payloads Extending Canonical Chain, Set Head to First Payload Received` requires reorgs 
* `Multiple New Payloads Extending Canonical Chain, Wait for Canonical Payload` (reorgs)
* `Valid NewPayload->ForkchoiceUpdated on Syncing Client` requires sync + p2p
* `Invalid Missing Ancestor ReOrg` (reorgs)
* `Invalid Missing Ancestor Syncing ReOrg` (reorg + sync)

Notes:

* The tests I added to the list above come from running the `engine` hive test suite with the "/cancun" filter. I went in order but had to skip some of the tests in the middle due to them using both V3 & V2 versions of the endpoints (when we only support V3). 
* I also tried to group test with the same prefix when possible, aka `NewPayloadV3 Versioned Hashes, Out of Order, Syncing=True ` & `NewPayloadV3 Versioned Hashes, Extra Hash, Syncing=True` both run under the pattern `NewPayloadV3 Versioned Hashes`)

Closes #issue_number

### CODE REVIEW

File: .github/workflows/hive.yaml [hunk 1]
```diff
@@ -24,12 +24,18 @@ jobs:
...
+          - simulation: rpc-auth
+            name: "Rpc Auth tests"
             run_command: make run-hive SIMULATION=ethereum/rpc-compat TEST_PATTERN="/engine-auth"
           - simulation: discv4
             name: "Devp2p discv4 tests"
-            run_command: make run-hive-debug SIMULATION=devp2p TEST_PATTERN="discv4"
+            run_command: make run-hive SIMULATION=devp2p TEST_PATTERN="discv4"
+          - simulation: engine
+            name: "Engine tests"
+            run_command:  make run-hive SIMULATION=ethereum/engine TEST_PATTERN="/Blob Transactions On Block 1, Cancun Genesis|Blob Transactions On Block 1, Shanghai Genesis|Blob Transaction Ordering, Single Account, Single Blob|Blob Transaction Ordering, Single Account, Dual Blob|Parallel Blob Transactions|ForkchoiceUpdatedV3 Modifies Payload ID on Different Beacon Root|NewPayloadV3 After Cancun|NewPayloadV3 Versioned Hashes|ForkchoiceUpdated Version on Payload Request"
```
Comment (mpaulucci) [inline .github/workflows/hive.yaml:35]: not sure I understand. Are these not cancun too?

File: crates/networking/rpc/engine/payload.rs [hunk 2]
```diff
@@ -32,9 +32,12 @@ impl RpcHandler for NewPayloadV3Request {
             return Err(RpcErr::BadParams("Expected 3 params".to_owned()));
         }
         Ok(NewPayloadV3Request {
-            payload: serde_json::from_value(params[0].clone())?,
-            expected_blob_versioned_hashes: serde_json::from_value(params[1].clone())?,
-            parent_beacon_block_root: serde_json::from_value(params[2].clone())?,
+            payload: serde_json::from_value(params[0].clone())
+                .map_err(|_| RpcErr::WrongParam("payload".to_string()))?,
```
Comment (mpaulucci) [inline crates/networking/rpc/engine/payload.rs:36]: do you think it makes sense yo have both `BadParams` and `WrongParam`? out of scope of this pr though

**Review by mpaulucci — APPROVED** [review]

File: .github/workflows/hive.yaml (hunk 1 above)
Comment (fmoletta) [reply .github/workflows/hive.yaml:35]: There are some tests that have two variants (Paris) and (Cancun), as they are not part of the name we cannot filter the cancun test variants (aka TEST_NAME (Cancun) is not valid), but filtering by `cancun/TEST_NAME` works well. I initially had these tests within the same pattern but it wasn't working in the CI (the cancun/ tests were being ignored), but it started working well once I split the pattern into two.
I also think the split is nice to keep CI times lower

File: crates/networking/rpc/engine/payload.rs (hunk 2 above)
Comment (fmoletta) [reply crates/networking/rpc/engine/payload.rs:36]: They have different error codes and hive expects different test cases to return different error codes so we have to comply with it

**Pull request merged**